algorithms.py

```diff
@@ -145,7 +178,100 @@ def configure(self, data, args):
         params = super(XgbGPUHistAlgorithm, self).configure(data, args)
         params.update({"tree_method": "gpu_hist", "gpu_id": 0})
         return params
+        
+class SkRandomForestAlgorithm(Algorithm):
+    def configure(self, data, args):
+        params = shared_params.copy()
+        if data.learning_task == LearningTask.REGRESSION:
+            params["objective"] = "regression"
```

Comment:
Let's reserve the `params` dictionary for actual hyperparameters that would be passed to Scikit-learn model constructor. Right now, only `max_depth` hyperparameter is being passed, so the `objective`, `scale_pos_weight` and `num_class` are all extraneous.